airbyte-cdk/python/unit_tests/sources/declarative/test_concurrent_declarative_source.py

```diff
@@ -739,37 +851,29 @@ def test_read_with_concurrent_and_synchronous_streams_with_sequential_state():
     palaces_records = get_records_for_stream("palaces", messages)
     assert len(palaces_records) == 7
 
-    # Expects 3 records, 1 empty slice, 3 records in slice
+    # Expects 3 records, 3 slices, 3 records in slice
     party_members_skills_records = get_records_for_stream("party_members_skills", messages)
-    assert len(party_members_skills_records) == 3
+    assert len(party_members_skills_records) == 9
 
 
-@freezegun.freeze_time("2024-09-10T00:00:00")
+@freezegun.freeze_time(_NOW)
 def test_read_concurrent_with_failing_partition_in_the_middle():
     """
     Verify that partial state is emitted when only some partitions are successful during a concurrent sync attempt
     """
 
+    most_recent_cursor_value = datetime(2024, 8, 10, 0, 0, 0, tzinfo=timezone.utc)  # based on _LOCATIONS_RESPONSE, this value might be outside of the actual start/end of the slice
+    # FIXME it seems like we yet don't consider `most_recent_cursor_value` as we ignore it during serialization [here](https://github.com/airbytehq/airbyte/blob/f07571f15f1bdbba86ad5e324e829a89b7d07cd6/airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py#L75-L78)
```

Comment:
@brianjlai aren't we supposed to rely on this before migrating low-code to the concurrent cursor?

Reply:
we discussed on slack and we need to investigate a little bit, but we might be okay on this since we're emitting as sequential state temporarily for this first release